mllib/src/main/scala/org/apache/spark/ml/Pipeline.scala

```diff
@@ -111,6 +111,18 @@ class Pipeline @Since("1.4.0") (
   @Since("1.2.0")
   def getStages: Array[PipelineStage] = $(stages).clone()
 
+  /** Returns stage at index i in Pipeline */
+  @Since("2.0.0")
+  def getStage[T <: PipelineStage](i: Int): T = getStages.apply(i).asInstanceOf[T]
```

Comment:
Here and elsewhere, I would just write array(i) instead of array.apply(i)

Also, can you please document the ```@tparam```?